crates/query-engine/translation/src/translation/query/values.rs

```diff
@@ -51,27 +51,31 @@ pub fn translate_json_value(
         _ => Ok(sql::ast::Expression::Cast {
             expression: Box::new(sql::ast::Expression::Cast {
                 expression: Box::new(Expression::Value(Value::JsonValue(value.clone()))),
-                r#type: sql::ast::ScalarType("jsonb".to_string()),
+                r#type: sql::ast::ScalarTypeName::new_unqualified("jsonb"),
             }),
             r#type: type_to_ast_scalar_type(r#type),
         }),
     }
 }
 
 /// Translate a NDC 'Type' to an SQL type name.
-fn type_to_ast_scalar_type(typ: &database::Type) -> sql::ast::ScalarType {
+fn type_to_ast_scalar_type(typ: &database::Type) -> sql::ast::ScalarTypeName {
     match typ {
         query_engine_metadata::metadata::Type::ArrayType(t) => {
-            // This will add multiple '[]'-suffixes when the input type represents a nested array.
-            // Postgresql does not have a notion of nested arrays at the type level, but accepts
-            // the array-of-arrays syntax for type names, simply interpreting it as though there
-            // were only a single pair of square brackets (e.g., 'int[][][]' simply becomes
-            // 'int[]'), which is what we want.
-            let scalar_type = type_to_ast_scalar_type(t).0;
-            sql::ast::ScalarType(scalar_type + "[]")
+            // This will collapse nested arrays. This is fine since it matches the behavior of
+            // Postgres where these are unsupported anyway.
+            let mut scalar_type = type_to_ast_scalar_type(t);
+            scalar_type.is_array = true;
+            scalar_type
+        }
+        query_engine_metadata::metadata::Type::ScalarType(t) => {
+            // TODO: This will need access to a mapping between ndc-type names and db type names
+            sql::ast::ScalarTypeName::new_unqualified(&t.0)
```

Comment:
`ast::ScalarTypeName` now supports schemas, but we do not track them yet from introspection.

This will change in subsequent PRs.